diff --git a/models/recommendation/pytorch/dlrm/product/dlrm_s_pytorch.py b/models/recommendation/pytorch/dlrm/product/dlrm_s_pytorch.py
index 3abfeedd..3d4e0d81 100644
--- a/models/recommendation/pytorch/dlrm/product/dlrm_s_pytorch.py
+++ b/models/recommendation/pytorch/dlrm/product/dlrm_s_pytorch.py
@@ -374,13 +374,14 @@ class DLRM_Net(nn.Module):

     def sequential_forward(self, dense_x, *emb_args):
         # process dense features (using bottom mlp), resulting in a row vector
-        x = self.apply_mlp(dense_x, self.bot_l)
+        #x = self.apply_mlp(dense_x, self.bot_l)
         # debug prints
         # print("intermediate")
         # print(x.detach().cpu().numpy())

         # process sparse features(using embeddings), resulting in a list of row vectors
         ly = self.apply_emb(self.emb_l, *emb_args)
+        return ly
         # for y in ly:
         #     print(y.detach().cpu().numpy())

